### PR TITLE
Add taint function and taints property to Node

### PR DESCRIPTION
**Pull request by thomasjpfan**
Addresses part of https://github.com/kr8s-org/kr8s/issues/326

This PR adds:

- A `taints` property, so `nodes.taints` gives you access to the taints.
- A `taint` function to add a taint
- A nice to have `effect="NoSchedule-"` that remove the taint. This is consistent with how `kubectl` works for removing a taint.

### CODE REVIEW

File: kr8s/_objects.py [hunk 1]
```diff
@@ -824,6 +826,33 @@ async def uncordon(self) -> None:
...
+                        "path": "/spec/taints",
+                        "value": [{"key": key, "value": value, "effect": effect}],
+                    }
+                ],
+                type="json",
+            )
+        else:
+            await self.async_patch(
+                {"spec": {"taints": [{"effect": effect, "key": key, "value": value}]}}
+            )
```
Comment (jacobtomlinson) [inline kr8s/_objects.py:846]: If there are already taints will this not replace them all with this new taint? We might need to do `"op": "add"` here instead.

Comment (thomasjpfan) [reply kr8s/_objects.py]: Good point.

Looking at `kubectl taint nodes ... my-key=value1:NoSchedule -v10`, I see it makes two calls, one to get all the taints, and another one update it. I updated this PR to copy what `kubectl` does.

File: kr8s/_objects.py [hunk 2]
```diff
@@ -828,6 +830,30 @@ async def uncordon(self) -> None:
         """
         await self.async_patch({"spec": {"unschedulable": False}})
 
+    async def taint(self, key: str, value: str, *, effect: str) -> None:
+        """Taint a node."""
```
Comment (jacobtomlinson) [inline kr8s/_objects.py:834]: Given that we are operating based on remote state we probably want to refresh the state before we continue.

```suggestion
        """Taint a node."""
        await self.async_refresh()
```

**Review by jacobtomlinson — APPROVED** [review]
Awesome thanks @thomasjpfan

**Pull request merged**